staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/conversion_test.go

```diff
@@ -526,3 +608,214 @@ func Test_ConvertJSONSchemaPropsToOpenAPIv2Schema(t *testing.T) {
 		})
 	}
 }
+
+// TestKubeOpenapiRejectionFiltering tests that the CRD openapi schema filtering leads to a spec that the
+// kube-openapi/pkg/util/proto model code support in version used in Kubernetes 1.13.
+func TestKubeOpenapiRejectionFiltering(t *testing.T) {
+	for i := 0; i < 10000; i++ {
+		t.Run(fmt.Sprintf("iteration %d", i), func(t *testing.T) {
+			f := fuzz.New()
+			seed := time.Now().UnixNano()
+			//seed = int64(1550678935445547778)
+			randSource := rand.New(rand.NewSource(seed))
+			f.RandSource(randSource)
+			t.Logf("seed = %d", seed)
+
+			fuzzFuncs(f, func(ref *spec.Ref, c fuzz.Continue, visible bool) {
+				var url string
+				if c.RandBool() {
+					url = fmt.Sprintf("http://%d", c.Intn(100000))
+				} else {
+					url = "#/definitions/test"
+				}
+				r, err := spec.NewRef(url)
+				if err != nil {
+					t.Fatalf("failed to fuzz ref: %v", err)
+				}
+				*ref = r
+			})
+
+			// create go-openapi object and fuzz it (we start here because we have the powerful fuzzer already
+			s := &spec.Schema{}
+			f.Fuzz(s)
+
+			// convert to apiextensions v1beta1
+			bs, err := json.Marshal(s)
+			if err != nil {
+				t.Fatal(err)
+			}
+			t.Log(string(bs))
+
+			var schema *apiextensionsv1beta1.JSONSchemaProps
+			if err := json.Unmarshal(bs, &schema); err != nil {
+				t.Fatalf("failed to unmarshal JSON into apiextensions/v1beta1: %v", err)
+			}
+
+			// convert to internal
+			internalSchema := &apiextensions.JSONSchemaProps{}
+			if err := apiextensionsv1beta1.Convert_v1beta1_JSONSchemaProps_To_apiextensions_JSONSchemaProps(schema, internalSchema, nil); err != nil {
+				t.Fatalf("failed to convert from apiextensions/v1beta1 to internal: %v", err)
+			}
+
+			// apply the filter
+			filtered, err := ConvertJSONSchemaPropsToOpenAPIv2Schema(internalSchema)
+			if err != nil {
+				t.Fatalf("failed to filter: %v", err)
+			}
+
+			// create a doc out of it
+			filteredSwagger := &spec.Swagger{
+				SwaggerProps: spec.SwaggerProps{
+					Definitions: spec.Definitions{
+						"test": *filtered,
+					},
+					Info: &spec.Info{
+						InfoProps: spec.InfoProps{
+							Description: "test",
+							Version:     "test",
+							Title:       "test",
+						},
+					},
+					Swagger: "2.0",
+				},
+			}
+
+			// convert to JSON
+			bs, err = json.Marshal(filteredSwagger)
+			if err != nil {
+				t.Fatalf("failed to encode filtered to JSON: %v", err)
+			}
+
+			// unmarshal as yaml
+			var yml yaml.MapSlice
+			if err := yaml.Unmarshal(bs, &yml); err != nil {
+				t.Fatalf("failed to decode filtered JSON by into memory: %v", err)
+			}
+
+			// create gnostic doc
+			doc, err := openapi_v2.NewDocument(yml, compiler.NewContext("$root", nil))
+			if err != nil {
+				t.Fatalf("failed to create gnostic doc: %v", err)
+			}
+
+			// load with kube-openapi/pkg/util/proto
+			if _, err := proto.NewOpenAPIData(doc); err != nil {
```

Comment:
question: the pipeline only tests if we've filtered out items that `gnostic` and `kube-openapi/pkg/util/proto` doesn't support, right? 

we haven't tested if the fuzzed Schema will end up being interpreted by kube-openapi/pkg/util/proto validator "too-strictly". maybe we don't have a way to test

Reply:
glad if we can reduce the validator divergence between server and client https://github.com/kubernetes/kube-openapi/pull/147#issuecomment-467215654

Reply:
> we haven't tested if the fuzzed Schema will end up being interpreted by kube-openapi/pkg/util/proto validator "too-strictly". maybe we don't have a way to test

Sounds like a good idea at first, but I am not sure we can create sensible documents validating against a fuzzed schema. Without fuzzing this does not make too much sense though.